xscen/regridding.py

```diff
@@ -52,105 +71,131 @@ def regrid(
       Regridded dataset
 
     """
+
     regridder_kwargs = regridder_kwargs or {}
 
-    # Whether or not regridding is required
-    if ds["lon"].equals(ds_grid["lon"]) & ds["lat"].equals(ds_grid["lat"]):
-        out = ds
-        if "mask" in out:
-            out = out.where(out.mask == 1)
-            out = out.drop_vars(["mask"])
+    ds_grids = []  # list of target grids
+    reg_arguments = []  # list of accompanying arguments for xe.Regridder()
+    if intermediate_reg_grids:
+        for name_inter, dict_inter in intermediate_reg_grids.items():
+            reg_arguments.append(dict_inter["regridder_kwargs"])
+            ds_grids.append(xesmf.util.cf_grid_2d(**dict_inter["cf_grid_2d"]))
+
+    ds_grids.append(ds_grid)  # add final ds_grid
+    reg_arguments.append(regridder_kwargs)  # add final regridder_kwargs
+
+    out = None
+    for i, (ds_grid, regridder_kwargs) in enumerate(zip(ds_grids, reg_arguments)):
+        # if this is not the first iteration (out != None),
+        # get result from last iteration (out) as input
+        ds = out or ds
+
+        # Whether or not regridding is required
+        if ds["lon"].equals(ds_grid["lon"]) & ds["lat"].equals(ds_grid["lat"]):
+            out = ds
+            if "mask" in out:
+                out = out.where(out.mask == 1)
+                out = out.drop_vars(["mask"])
```

Comment:
See next comment regarding dropping `mask`.

Reply:
Hum en effet, je n'y avais pas penser. Je vais mettre le if ''# Whether or not regridding is required'' en dehors du loop. Si ds et ds_grid ont les mêmes lat et lon, on ne devrait rien faire, même si des intermediate_grids ont été donnée.